advanced/index.php

```diff
@@ -19,15 +19,20 @@ function validIP($address){
 // Define which URL extensions get rendered as "Website Blocked"
 $webExt = array('asp', 'htm', 'html', 'php', 'rss', 'xml');
 
+// Get IPv4 and IPv6 addresses from setupVars.conf (if available)
+$setupVars = parse_ini_file("/etc/pihole/setupVars.conf");
+$ipv4 = isset($setupVars["IPV4_ADDRESS"]) ? explode("/", $setupVars["IPV4_ADDRESS"])[0] : $_SERVER['SERVER_ADDR'];
+$ipv6 = isset($setupVars["IPV6_ADDRESS"]) ? explode("/", $setupVars["IPV6_ADDRESS"])[0] : $_SERVER['SERVER_ADDR'];
+
 $AUTHORIZED_HOSTNAMES = array(
 	$ipv4,
 	$ipv6,
-	str_replace(array("[","]"), array("",""), $_SERVER["SERVER_NAME"]),
+	str_replace(array("[","]"), array("",""), $_SERVER["SERVER_ADDR"]),
```

Comment:
if authorized hostnames is populated with `$_SERVER["SERVER_NAME"]` instead of `$_SERVER["SERVER_ADDR"]`, the blocking page will never be shown since the second test in line https://github.com/pi-hole/pi-hole/pull/1397/files#diff-622eb992a761acc873605fcdfbaf1e3bR39 will always be true.